### ISSUE TITLE
CT: Which GI Bill benefit do you want to use? (Learn more) resets to Post-9/11 on profile page

### ISSUE THREAD

**Issue by sonntag-adam** [department-of-veterans-affairs/va.gov-team#8228]
Steps to reproduce: 
**Open browser in incognito mode**

1. Copy and paste https://staging.va.gov/gi-bill-comparison-tool, **don't** select the link
2. For "Which GI Bill benefit do you want to use? (Learn more)" select Montgomery GI Bill
3. Search on College of Charleston and select school on the search results page
4. Observe that  "Which GI Bill benefit do you want to use? (Learn more)" is back to Post-9/11 GI Bill (Ch 33) on the profile page.

**Comment by dmccurdybah** [comment]
Unable to reproduce using steps on prod, staging or local.

> You can also select another benefit on the profile page that is not Post-9/11 GI Bill (Ch 33) and when refreshing the profile page it will reset back to Post-9/11 GI Bill (Ch 33) by default.

This is because the application state is reset by reloading the page. This is true of all functionality in the comparison tool.

This is related to the issue that I think is confusing testing for https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/7528

**Comment by desireeturner** [comment]
Moving to validate to confirm that this cannot be reproduced with additional details or context @sonntag-adam. Might be something we want @department-of-veterans-affairs/uxbah to review too.

**Comment by amyknox** [comment]
I can reproduce the issue of the benefit re-setting to Post-9/11 as you move from Landing Page to Profile Page on both staging and prod. Looks like it actually resets from SRP to Profile. 

Wonder if it's a PC/Mac thing?

**Comment by dmccurdybah** [comment]
I'm able to reproduce this in staging.

1) In Chrome in a new incognito, go to https://staging.va.gov/gi-bill-comparison-tool/search?category=school&name=charleston
2) Change benefit from Chapter 33
3) Select a search result

**Comment by ksantiagomorgan6** [comment]
pull request: 

https://github.com/department-of-veterans-affairs/vets-website/pull/12525

**Comment by sonntag-adam** [comment]
Still occurs in staging

**Comment by ksantiagomorgan6** [comment]
New Pull request 5-8-2020:

https://github.com/department-of-veterans-affairs/vets-website/pull/12598

**Comment by sonntag-adam** [comment]
Still occurs in staging after merge. Moving back to in progress

**Comment by sonntag-adam** [comment]
https://github.com/department-of-veterans-affairs/vets-website/pull/12609

**Comment by dmccurdybah** [comment]
PR: https://github.com/department-of-veterans-affairs/vets-website/pull/12630

**Comment by sonntag-adam** [comment]
https://github.com/department-of-veterans-affairs/vets-website/pull/12647

**Comment by sonntag-adam** [comment]
https://github.com/department-of-veterans-affairs/vets-website/pull/12649

**Comment by sonntag-adam** [comment]
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/48804582/81971205-b8003d80-95ee-11ea-9b74-c6d5d0ce0cb5.gif)

**Comment by sonntag-adam** [comment]
Still occurs in staging.

**Comment by dmccurdybah** [comment]
Another PR: https://github.com/department-of-veterans-affairs/vets-website/pull/12889

**Comment by sonntag-adam** [comment]
Testing passes locally

**Comment by sonntag-adam** [comment]
Still occurs in staging.

**Comment by dmccurdybah** [comment]
Additional PR to remove unnecessary react-router `with-router` usage:

https://github.com/department-of-veterans-affairs/vets-website/pull/13096

**Comment by sonntag-adam** [comment]
Still occurs in staging

**Comment by desireeturner** [comment]
Waiting on VSP team to make an update to the react-router library sometime soo and we can see if that fixes this.

**Comment by amyknox** [comment]
@dmccurdybah - is raising awareness of this to platform. If we don't get traction, address at 9/1 ToTs

**Comment by WillMcCormack** [comment]
Bringing this up on Todays' ToTs

**Comment by dmccurdybah** [comment]
Rian Fowler did give me an update yesterday on using the new version of react-router but I haven't had a chance to look into it

> It’s not on the roadmap for the forms library apps but we’re using v5 on the profile page and vaos. There’s no documentation for this so you’ll probably have to look at those apps to figure out how to use it on a non-form app.

That's fine for us because GIDS doesn't use the forms library apps.

**Comment by dmccurdybah** [comment]
PR for react-router-dom update: https://github.com/department-of-veterans-affairs/vets-website/pull/14143

**Comment by sonntag-adam** [comment]
Still occurs in staging.

**Comment by dmccurdybah** [comment]
PR: https://github.com/department-of-veterans-affairs/vets-website/pull/15023

**Comment by dmccurdybah** [comment]
PR: https://github.com/department-of-veterans-affairs/vets-website/pull/15052

**Comment by sonntag-adam** [comment]
Testing passes QA
